arbnode/transaction_streamer.go

```diff
@@ -1653,6 +1692,96 @@ func (s *TransactionStreamer) submitEspressoTransactions(ctx context.Context, ig
 	return s.config().EspressoTxnsPollingInterval
 }
 
+func (s *TransactionStreamer) toggleEscapeHatch(ctx context.Context) error {
+	live, err := s.lightClientReader.IsHotShotLive(s.config().EspressoSwitchDelayThreshold)
+	if err != nil {
+		return err
+	}
+	// If hotshot is down, escape hatch is activated, the only thing is to check if hotshot is live again
+	if s.escapheHatchOn {
+		if live {
+			log.Info("HotShot is up, disabling the escape hatch")
+			s.escapheHatchOn = false
+		}
+		return nil
+	}
+
+	// If hotshot is up, escape hatch is disabled
+	// - check if escape hatch should be activated
+	// - check if the submitted transaction should be skipped from espresso verification
+	if !live {
+		log.Warn("enabling the escape hatch, hotshot is down")
+		s.escapheHatchOn = true
+	}
```

Comment:
`escapeHatchOn` is only used to check if the batch poster should submit transactions to hotshot. If hotshot is `currently` down, then bypass it. If hotshot is `currently` up, then turn off the escape hatch. Don't we need a real time liveness to see if we should enable/disable the escape hatch?

Reply:
I think using the L1 height in the message header is incorrect. HotShot blocks also have an L1 height. During the time when transaction approaching HotShot, the L1 height can be different.

Reply:
@zacshowa 


One thing to note here:
```
Say we have a message with l1 height x where hotshot was live based on the data stored in the light client
```

This doesnt apply anymore because earlier we had changes in the sequencer code, so we decided if we want to create an espresso message or not based on whether hotshot was live. But now, all we care about is that hotshot is able to include the l2Messages before it gets posted to L1 in real time. So I think Jeremy's current code behaviour seems right but let me know if you think otherwise!